--- origsrc/gst-plugins-good-0.10.8/configure.ac	2008-04-22 20:55:55.000000000 -0500
+++ src/gst-plugins-good-0.10.8/configure.ac	2008-06-19 21:54:29.890625000 -0500
@@ -327,7 +327,8 @@
   AG_GST_DISABLE_PLUGIN(rtsp)
 fi
 
-if test "x$HAVE_WINSOCK2_H" = "xyes"; then
+dnl Cygwin has both, but only sys/socket.h should be used
+if test "x$HAVE_WINSOCK2_H" = "xyes" -a "x$HAVE_SYS_SOCKET_H" != "xyes" ; then
   WIN32_LIBS="-lws2_32"
   AC_SUBST(WIN32_LIBS)
 fi
--- origsrc/gst-plugins-good-0.10.8/tests/check/Makefile.am	2008-04-16 00:24:13.000000000 -0500
+++ src/gst-plugins-good-0.10.8/tests/check/Makefile.am	2008-06-19 21:54:29.906250000 -0500
@@ -102,8 +102,8 @@
 
 # these tests don't even pass
 # autodetect: temporarily disabled because of broken videosinks in -bad
-noinst_PROGRAMS = \
-	elements/autodetect
+# noinst_PROGRAMS = \
+#	elements/autodetect
 
 AM_CFLAGS = $(GST_OBJ_CFLAGS) $(GST_CHECK_CFLAGS) $(CHECK_CFLAGS) $(GST_OPTION_CFLAGS)
 LDADD = $(GST_OBJ_LIBS) $(GST_CHECK_LIBS) $(CHECK_LIBS)
